client/src/components/Q&A/QuestionModal.jsx

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import styling from './questions.css';
+
+const QuestionModal = (props) => {
+  const [productName, setProductName] = useState('');
+  const [textInput, setTextInput] = useState('');
+  const [nameInput, setNameInput] = useState('');
+  const [emailInput, setEmailInput] = useState('');
+  const [textCheck, setTextCheck] = useState(false);
+  const [nameCheck, setNameCheck] = useState(false);
+  const [emailCheck, setEmailCheck] = useState(false);
+
+  // Destructuring
+  const { toggleQuestionForm, productId } = props;
+
+  // Input change handlers
+  const textChangeHandler = (event) => {
+    setTextInput(event.target.value);
+    if (event.target.value.length > 0) {
+      setTextCheck(true);
+    } else {
+      setTextCheck(false);
+    }
+  };
+
+  const nameChangeHandler = (event) => {
+    setNameInput(event.target.value);
+    if (event.target.value.length > 0) {
+      setNameCheck(true);
+    } else {
+      setNameCheck(false);
+    }
+  };
+
+  const emailChangeHandler = (event) => {
+    setEmailInput(event.target.value);
+    if (event.target.value.indexOf('@') > -1) {
+      setEmailCheck(true);
+    } else {
+      setEmailCheck(false);
+    }
+  };
+
+  // submit question information to API
+  const submitQuestion = () => {
+    const questionInfo = {
+      body: textInput,
+      name: nameInput,
+      email: emailInput,
+      product_id: productId,
+    };
+    if (textCheck && nameCheck && emailCheck) {
+      axios.post('/qa/questions', questionInfo)
+        .then(() => {
+          toggleQuestionForm();
+        })
+        .catch((error) => {
+          console.log('error in posting question on the client', error);
+        });
+    } else {
+      alert('You must contain the following: answer, nickname, and email.');
+    }
+  };
+
+  // Get product name
+  const getProductName = () => {
+    axios.get(`/products/${productId}`)
+      .then((response) => {
+        setProductName(response.data.name);
+      })
+      .catch((error) => {
+        console.log('error in getting product name on the client side', error);
+      });
+  };
+
+  useEffect(getProductName, [productId]);
+
+  return (
+    <div className="modal">
+      <div className="modal-content">
+        <div className="modal-header">
+          <h4 className="modal-title">Ask Your Question</h4>
+          <span>
+            &#34;
+            About the&#160;
+            {productName}
+            &#34;
+          </span>
+        </div>
+        <form>
+          <div className="modal-body">
+            <span>Your Question:</span>
```

Comment:
Also great job on the Question Modal! There also seems to be a part in the business requirements that would like an asterisk next to the title if a fill out is mandatory.

"...Each should be labelled as titled below. Those indicated as mandatory should have an asterisk next to the title."
1.3.5.1. Your Question (mandatory)
1.3.5.2. What is your nickname (mandatory)
1.3.5.3. Your email (mandatory)

Reply:
edited